types/dec_coin.go

```diff
@@ -478,6 +478,27 @@ func (coins DecCoins) IsAllPositive() bool {
 	return true
 }
 
+// MinSet will return a new set of coins which contains the minimum decCoin
+// for common denoms found in both `coins` and `coinsB`. For denoms not common
+// to both `coins` and `coinsB` the minimum is considered to be 0, thus they are
+// not added to the final set.
+func MinSet(coinsA, coinsB DecCoins) DecCoins {
```

Comment:
What about a more general function signature like:

```go
func (d DecCoins) MinSet(compCoins ...DecCoins) DecCoins {}
```

Reply:
meh - typically we use `MinDec`/`MinInt` to check if it overflows the available amount (which is what we are using this for as well. I think until there is a clear use case, it's easiest to just keep it as is - implementing this suggested feature will increase the complexity of the code

Reply:
This doesn't remove zero coins, and it's already implemented as coins.Cap(coinsB).
I'll update to use Cap, but rename Cap is Intersect.  I think Intersect is better than Cap or MinSet, as it's the mathematical definition of "intersection" of two sets of coins, which is what both functions do.  coinsA.Intersect(coinsB) so it will have the above the signature, (aka is a method, not just a module function).

Reply:
Agreed on `Intersect`.

Reply:
I will implement @jaekwon's suggestion as per sdk meeting discussion